web/src/lib/components/shared-components/version-announcement-box.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+	import { getGithubVersion } from '$lib/utils/get-github-version';
+	import { onMount } from 'svelte';
+	import FullScreenModal from './full-screen-modal.svelte';
+	import type { ServerVersionReponseDto } from '@api';
+
+	export let serverVersion: ServerVersionReponseDto;
+
+	let showModal = false;
+	let githubVersion: string;
+	$: serverVersionName = semverToName(serverVersion);
+
+	function semverToName({ major, minor, patch }: ServerVersionReponseDto) {
+		return `v${major}.${minor}.${patch}`;
+	}
+
+	function onAcknowledge() {
+		// Store server version to prevent the notification
+		// from showing again.
+		localStorage.setItem('appVersion', serverVersionName);
+		showModal = false;
+	}
+
+	onMount(async () => {
+		try {
+			if (localStorage.getItem('appVersion') === serverVersionName) {
+				// Updated version has already been acknowledged.
+				return;
+			}
+
+			githubVersion = await getGithubVersion();
+			if (githubVersion !== serverVersionName) {
+				showModal = true;
+			}
```

Comment:
I think this logic is not quite correct. If the server is on, say, 1.49.0, and that version has been acknowledged and stored locally, this check will return before ever comparing to the githubVersion.

Reply:
Good catch! It would be correct for only showing the notification once after the server gets outdated. But I've fixed it and the notification will now show for every update.